### ISSUE TITLE
GLFW 3.4 support?

### ISSUE THREAD

**Issue by TheOnlySilverClaw** [c3lang/vendor#25]
I just saw you were working on GLFW bindings, but only for version 3.3

I actually made my own 3.4 bindings already:
https://github.com/TheOnlySilverClaw/glfw.c3l

Just wanted to ask whether you're also planning to support 3.4 anytime soon, since I need functions like `glfwGetPlatform`.

**Comment by lerno** [comment]
Feel free to add a pull request to support 3.4 as well. You can look at how this is done with opengl to support different versions.

**Comment by TheOnlySilverClaw** [comment]
I'll have a look. I've also noticed the current bindings don't make use of receiver functions, submodules or anything "fancy", but rather do 1:1 mappings. Is that a deliberate choice?

Just asking because I did that in my version, for example:
https://github.com/TheOnlySilverClaw/glfw.c3l/blob/develop/window.c3i
and would have to change that accordingly.

**Comment by lerno** [comment]
Yes, I think we should probably think of bindings as a two step process:
1. The full bindings, trying to match things 1:1 so that people can get started just following along with examples written in C and just port them without having to look up the C3 names and helper functions.
2. A custom wrapper / binding, which builds on (1) but adds C3 conveniences and can depart arbitrarily far from the "strict" bindings.

So step (1) is always the main priority, and then (2) doesn't necessarily need to be in vendor but could be offered by people from their own repos.

There is an idea for a search where people could register their libraries, and people could search them. Here one could then have several such bindings of style (2) to offer people whatever style they prefer.

Search could get connected to the vendor-fetch functionality as well, allowing as convenient a fetch as vendor would.

**Comment by TheOnlySilverClaw** [comment]
Makes sense. If you want, you can have a look at my version. I've tried adding some namespacing, but other than that, it's nearly 1:1 and pretty complete.

If that's already too much deviation for your taste, I'll see whether I can adapt the 3.4 functions to the vendor style and long-term use those to build my own wrapper around.

**Comment by lerno** [comment]
I think that eventually, we'll probably have to have someone who goes through all the bindings and make sure they're consistent in style, but that's more as we go towards 0.8, 0.9 thereabouts, for now it's important that people just have bindings so that they can get started quickly. If your style isn't compatible with the other GLFW then you could have a distinct name, for example it could be "glfw34.c3l" and then you'd not need to sync it with the other glfw that we have. Does that sound good to you?

**Comment by TheOnlySilverClaw** [comment]
I think I'll just use and maintain mine separately for now.
